### PR TITLE
DEV: Full system specs coverage for signup/login

### PR DESCRIPTION
**Pull request by jancernik**

### CODE REVIEW

**Review by pmusaraj — COMMENTED** [review]
This looks good as is to me (maybe just missing some cleanups). Nice!

File: spec/support/omniauth_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      uid: "12345",
+      info:
+        OmniAuth::AuthHash::InfoHash.new(
+          email: email,
+          first_name: first_name,
+          last_name: last_name,
+        ),
+    )
+
+    Rails.application.env_config["omniauth.auth"] = OmniAuth.config.mock_auth[:linkedin_oidc]
```
Comment (pmusaraj) [inline spec/support/omniauth_helpers.rb:80]: Maybe best to add a method here for `def reset_omniauth_config` and use that, so it's clearer to people consuming these in the future that they need to be reset after every test.

**Review by pmusaraj — APPROVED** [review]
Overall looks good to me, left some minor comments. We can merge and fine-tune though.

File: spec/system/login_spec.rb [hunk 2]
```diff
@@ -30,18 +38,45 @@
...
+      find("form .btn-primary").click
+      expect(page).to have_css(".header-dropdown-toggle.current-user")
     end
   end
 
-  context "with magic link" do
+  context "with login link" do
     it "can login" do
-      # TODO: add test
+      Jobs.run_immediately!
```
Comment (pmusaraj) [inline spec/system/login_spec.rb:65]: Might be useful to move `Jobs.run_immediately!` to a before block. It's helpful on all tests in this file.

File: spec/system/social_authentication_spec.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    it "works" do
+      mock_twitter_auth
+      visit("/")
+
+      login_modal.open
+      login_modal.select_twitter
+      expect(signup_modal).to be_open
+      expect(signup_modal).to have_no_password_input
+      signup_modal.fill_email(OmniauthHelpers::EMAIL)
```
[the remark this answers is not in the document]
Comment (jancernik) [reply spec/system/social_authentication_spec.rb:83]: It doesn't return an email, no

https://github.com/arunagw/omniauth-twitter?tab=readme-ov-file#authentication-hash

**Review by ZogStriP — APPROVED** [review]
Love it 😍 

Those specs are a work of art 👨‍🎨

File: spec/system/login_spec.rb [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      login_modal.open
+      login_modal.fill(username: "john", password: "supersecurepassword")
+      login_modal.click_login
+      find(".activation-controls button.resend").click
+
+      wait_for(timeout: 5) { ActionMailer::Base.deliveries.count != 0 }
+
+      mail = ActionMailer::Base.deliveries.last
+      expect(mail.to).to contain_exactly(user.email)
+      activation_link = mail.body.to_s[%r{/u/activate-account/\S+}, 0]
```
[the remark this answers is not in the document]
Comment (jancernik) [reply spec/system/login_spec.rb:31]: Yes, you're right. The first match is the default

**Pull request merged**